activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -576,15 +576,18 @@ def find_last(limit)
       end
 
       def ordered_relation
-        if order_values.empty? && (implicit_order_column || primary_key)
-          if implicit_order_column && primary_key && implicit_order_column != primary_key
-            order(table[implicit_order_column].asc, table[primary_key].asc)
-          else
-            order(table[implicit_order_column || primary_key].asc)
-          end
-        else
-          self
-        end
+        return self unless order_values.empty? && (implicit_order_column || !query_constraints_list.empty?)
```

Comment:
I generally like early returns but I'm finding this one harder to read than the previous implementation. I think I'd rather this use the if/else than the early return.

Reply:
Fair, I brought the `if` `else` back 👍